### DISCUSSION TITLE
examples/chat-persistent.sh - does it work?

### DISCUSSION THREAD

**Discussion by quarterturn** [ggml-org/llama.cpp#1939] [Q&A]
I noticed there's a 'examples/chat-persistent.sh' which would appear to try to provide a sliding window for tokens. It would be fun to be able to maintain a chat between sessions and chat for more than 2048 tokens. The question is, does it work for anyone? The same prompt which works fine in normal interactive mode generates an offtopic, confused mess with chat-persistent.sh.

**Comment by ejones** [comment]
Hey, I wrote this script and have used it frequently since. Note that I've only ever used it with the base LLaMA models. Are you still experiencing this? Do you mind sharing your prompt, model, and any other relevant details?
